docs/adapters/development.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: docs
+permalink: /docs/adapters/development/index.html
+---
+
+# Adapter Basics
+
+All adapters inherit from the Adapter class in the `src/adapter.coffee` file.  There are certain methods that you will want to override.  Here is a basic stub of what an extended Adapter class would look like:
+
+```coffee
+class Sample extends Adapter
+
+  constructor: ->
+    @robot.logger.info "Constructor"
+    super
+    
+
+  send: (envelope, strings...) ->
+    @robot.logger.info "Send"
+
+  reply: (envelope, strings...) ->
+    @robot.logger.info "Reply"
+
+  run: ->
+    @robot.logger.info "Run"
+    @emit "connected"
```

Comment:
Could we include an example of receiving a message here? It can be hard coded, but with a comment that you'd normally connect to the service, and start parsing messages and sending them over to receive.